mongodb-nosqli.yaml

```diff
@@ -0,0 +1,109 @@
+rules:
+- id: mongodb-nosqli
+  message: Detected non-constant data passed into a NoSQL query using the 'where'
+    evaluation operator. If this data can be controlled by an external user, this
+    is a NoSQL injection. Ensure data passed to the NoSQL query is not user controllable,
+    or properly sanitize the data. Ideally, avoid using the 'where' operator at all
+    and instead use the helper methods provided by com.mongodb.client.model.Filters
+    with comparative operators such as eq, ne, lt, gt, etc.
+  languages:
+  - java
+  severity: WARNING
+  patterns:
+  - pattern-either:
+    - pattern: (BasicDBObject $QUERY).put("$where", $INPUT);
```

Comment:
I would replace all class names with their fully qualified name to get more accurate findings.
In your case it would simply be `BasicDBObject -> com.mongodb.BasicDBObject` and `BasicDBObjectBuilder -> com.mongodb.BasicDBObjectBuilder`.

Reply:
sounds good

Reply:
Fixed in 67842403b5064740657a5810ea9576bc15f1f84c